### PR TITLE
2 get topics

### PR DESCRIPTION
**Pull request by simonhighfield**

### CODE REVIEW

**Comment by simonhighfield** [comment]
Hi, I left a comment in app.js asking respond.status ... vs res.status ... 
The question is whre is that defined?
I think it's the input parameter to that function but some clarity would be greatly apreciated

File: MVC/controller.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
-const {fetch} = require("./models")
+const {fetchTopics} = require("./models")
 
-exports. get = (req, res, next) => {
+exports. getTopics = (req, res, next) => {
     // console.log('controller.js bounjour');
-    fetch()
+    fetchTopics()
+    .then((topics) => {
+        res.status(200).send(topics)
```
Comment (nc-saima-khanom) [inline MVC/controller.js:6]: We should send back data to the client on a meaningful key e.g, they should be able to access it on body.topics - at the moment we are sending an object through but not on a key of topics - how could we ensure the user can access the topics array on a key of topics? :)

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -7,8 +7,19 @@ const app = require('../MVC/app')
...
+        
+    test('GET200: endpoint that responds with all topics with slug and description properties', () => {
+        return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({body})=>{
+            const topics = body
+            expect(topics.length).toBe(testData.topicData.length)
+            topics.forEach((topic)=>{
+                expect(typeof topic.slug).toBe("string")
```
Comment (nc-saima-khanom) [inline __tests__/integration.test.js:33]: Another matcher that may be useful to look into: .toMatchObject()

File: __tests__/integration.test.js [hunk 3]
```diff
@@ -7,8 +7,19 @@ const app = require('../MVC/app')
 beforeEach(() => { return seed(testData) })
 afterAll(() => db.end())
 
-describe('GET/api/topics', () => {
-    test('check this runs', () => {
-
+describe('GET: /api/topics', () => {
```
Comment (nc-saima-khanom) [inline __tests__/integration.test.js:24]: We can also test for any requests to a non-existent endpoint or set up error handling middleware too! :D

**Pull request merged**